Assets/Scripts/Bomb.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Basic enemy that simply approaches player when player enters its awareness radius.
+/// </summary>
+/// <remarks>
+/// Placeholder enemy whose graphics are stolen from Final Fantasy VII Remake
+/// </remarks>
+public class Bomb : Enemy {
+    /// <summary>
+    /// Initial value for maximum speed at which Bomb can approach the player
+    /// </summary>
+    /// <remarks>
+    /// If you wish to tweak Bomb's max speed in the Inspector at runtime, alter max_speed variable from _awareBehaviour and _engagedBehaviour instead.
+    /// </remarks>
+    private const float _MAX_SPEED = 4;
+    /// <summary>
+    /// Initial value for magnitude of acceleration with which Bomb can approach the player
+    /// </summary>
+    /// <remarks>
+    /// If you wish to tweak Bomb's acceleration rate in the Inspector at runtime, alter acceleration_rate variable from _awareBehaviour and _engagedBehaviour instead.
+    /// </remarks>
+    private const float _ACCELERATION_RATE = 1;
+
+    /// <summary>
+    /// Player's Rigidbody2D
+    /// </summary>
+    private Rigidbody2D _player;
+
+    /// <summary>
+    /// Sets instance variables such that Bomb does nothing when idle and approaches player otherwise
+    /// </summary>
+    protected override void Start() {
+        this._myRigidbody = this.GetComponent<Rigidbody2D>();
+        this._mySprite = this.GetComponent<SpriteRenderer>();
+        this._player = GameObject.FindGameObjectWithTag("Player").GetComponent<Rigidbody2D>();
+        this._behaviours.Add(ENEMY_STATE.idle, (new DoNothing(this, ENEMY_STATE.idle), AfterIdleAnimate));
+        this._behaviours.Add(ENEMY_STATE.aware, (new ApproachTarget(this, this._player, Bomb._MAX_SPEED, Bomb._ACCELERATION_RATE, ENEMY_STATE.aware), AfterAwareAnimate));
+        this._behaviours.Add(ENEMY_STATE.engaged, (new ApproachTarget(this, this._player, Bomb._MAX_SPEED, Bomb._ACCELERATION_RATE, ENEMY_STATE.engaged), AfterEngagedAnimate));
```

Comment:
The consequence of how I changed the dictionary structure is now adding elements to the dictionary is kinda complicated. I'd make a helper function, but the only thing it could do without itself using a switch to decide which animation function to use based on the ENEMY_STATE is to make it so you can pass the ENEMY_STATE value, the EnemyBehaviour, and the animation function as three separate parameters instead of an ENEMY_STATE value and a tuple of the other two parameters. Which means you call a helper function so you don't have to write two parentheses. Is that even worth it?

Reply:
We could create a new class that serves as a data structure for this pattern

Reply:
Are you suggesting we make a wrapper class `EnemyData` that stores an `EnemyBehaviour` and an animation function (probably in the form of an `Action`)? And then make `behaviours` a `Dictionary<ENEMY_STATE, EnemyData>`, and in individual enemy classes like `Bomb` we split up the process of adding an element to `behaviours` across two lines, say

`EnemyData idleData = new EnemyData(new DoNothing(this, ENEMY_STATE.idle), AfterIdleAnimate);
this._behaviours.Add(ENEMY_STATE.idle, idleData);`

for greater code readability?

Reply:
Aye

Reply:
Since we had this conversation, we refactored Bomb.cs Start method to make the use of the structure significantly more readable, and we've made minor tweaks to the struct's API to make much more convenient to use.

As I have gotten the approval I so deeply craved, I'm resolving this.